README.md

```diff
@@ -101,6 +101,44 @@ public class Main {
 }
 ```
 
+## Reusing http clients
+
+Make sure you don't create a new provider every time you want to use one. Instead, you should reuse existing instance.
+This way you reuse connections and thread pools. You can read more about it [here](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/#okhttpclients-should-be-shared).
+
+✅ **Do:** 
+```java
+var provider = GatewayProvider.makeTestnetClient();
+var account1 = new StandardAccount(provider, accountAddress1, privateKey1);
+var account2 = new StandardAccount(provider, accountAddress2, privateKey2);
+```
+
+❌ **Don't:**
+```java
+var provider1 = GatewayProvider.makeTestnetClient();
+var account1 = new StandardAccount(provider1, accountAddress1, privateKey1);
+var provider2 = GatewayProvider.makeTestnetClient();
+var account2 = new StandardAccount(provider2, accountAddress2, privateKey2);
+```
+
+Alternatively you can create a `HttpService` yourself and pass it whenever creating a provider. This way your whole 
+application can use a single `OkHttpClient`. 
+
+```java
+import com.swmansion.starknet.service.http.OkHttpService;
+
+// ...
+
+var httpClient = new OkHttpClient();
+var httpService = new OkHttpService(httpClient);
+
+var provider = GatewayProvider.makeTestnetClient(httpService);
+var account1 = new StandardAccount(provider, accountAddress1, privateKey1);
+var account2 = new StandardAccount(provider, accountAddress2, privateKey2);
```

Comment:
We use explicit types in other places.

Reply:
(the code above can be copied to an actual codebase, code here doesn't make sense apart from illustrative reasons)